### PR TITLE
ci: eslint rule enforce-module-boundaries

### LINKED ISSUE #1260: 🤖 - Investigate cyclic dependencies
### Description

- https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-cycle.md
- https://www.typescriptlang.org/tsconfig#importsNotUsedAsValues

### PR DESCRIPTION
**Pull request by mullinsmikey**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [x] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #1260

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -38,5 +39,32 @@ module.exports = {
         ],
         'no-return-assign': ['error', 'always'],
         'no-implicit-coercion': ['error', {allow: ['!!']}],
+
+        '@nrwl/nx/enforce-module-boundaries': [
+            'warn', // TODO: increase error level as soon as warnings are fixed
```
Comment (splincode) [inline .eslintrc.js:44]: Hello, for some reason there are no warnings in the logs, is it correct?

image

Comment (mullinsmikey) [reply .eslintrc.js:44]: Definitely not. Please check that you have a clean copy of the branch, run `npm ci` and restart the Nx daemon with `nx reset`. After killing the daemon this rule might not work at all until you perform any build, e.g. with `npm start`. 
There should be over 200 warnings with this rule.

Comment (splincode) [reply .eslintrc.js:44]: `200 warnings` are you sure they are problems?

Comment (mullinsmikey) [reply .eslintrc.js:44]: I'm pretty sure that there's a lot to consider. E.g. we have a lot of imports from the `demo` in `addon-doc` tests.
Of course I didn't do manual checks for every warning but looked into a couple of them and at least 20-30% of these should be fixed.

Comment (splincode) [reply .eslintrc.js:44]: Thank you for explain!

**Review by waterplea — APPROVED** [review]

**Pull request merged**